ci/cpp_linters.sh

```diff
@@ -0,0 +1,47 @@
+#!/bin/bash
+# Copyright (c) 2024, NVIDIA CORPORATION.
+
+set -euo pipefail
+
+rapids-logger "Create checks conda environment"
+. /opt/conda/etc/profile.d/conda.sh
+
+ENV_YAML_DIR="$(mktemp -d)"
+
+rapids-dependency-file-generator \
+  --output conda \
+  --file-key clang_tidy \
+  --matrix "cuda=${RAPIDS_CUDA_VERSION%.*};arch=$(arch);py=${RAPIDS_PY_VERSION}" | tee "${ENV_YAML_DIR}/env.yaml"
+
+rapids-mamba-retry env create --yes -f "${ENV_YAML_DIR}/env.yaml" -n clang_tidy
+
+# Temporarily allow unbound variables for conda activation.
+set +u
+conda activate clang_tidy
+set -u
+
+RAPIDS_VERSION_MAJOR_MINOR="$(rapids-version-major-minor)"
+
+source rapids-configure-sccache
+
+# TODO: For testing purposes, clone and build IWYU. We can switch to a release
+# once a clang 19-compatible version is available, which should be soon
+# (https://github.com/include-what-you-use/include-what-you-use/issues/1641).
+git clone https://github.com/include-what-you-use/include-what-you-use.git
```

Comment:
```suggestion
git clone --depth 1 https://github.com/include-what-you-use/include-what-you-use.git
```

You only need the tip of this, right? A `--depth-1` would make this clone a little faster (and maybe less susceptible to network issues, because you'd be pulling a smaller amount of data over the wire).

Reply:
Sure we might as well. Normally I'd be more enthusiastic, but in this case hopefully this clone will go away altogether very soon :) so not as big a deal. No harm improving this in the meantime though.